src/pages/Counter/containers/CounterContainer.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from "react";
+import PropTypes from 'prop-types';
+
+import { CounterView } from "../components/CounterView/index.js";
+
+class CounterComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      countValue: 0,
+      parityType: "even",
+    };
+  }
+
+  calculateParityType = (state) => {
+    return state.countValue % 2 === 0 ? "even" : "odd";
+  };
+
+  handleIncrement = () => {
+    this.setState((state) => {
+      return {
+        countValue: state.countValue + 1,
+      };
+    });
+    this.setState((state) => {
+      return {
+        parityType: this.calculateParityType(state),
+      };
```

Comment:
you don't need this `setState`. you can do the same in the `setState` above

Reply:
Move parity type to componentDidUpdate()